packages/lesswrong/components/common/FormatDate.tsx

```diff
@@ -14,10 +14,38 @@ const formatRelative = (
   now: Date,
   includeAgo?: boolean,
 ): string => {
-  const formatted = moment(date).from(now);
+  //const formatted = formatRelativeMoment(new Date(date), now);
+  const formatted = formatRelativeFast(new Date(date), now);
   return includeAgo && formatted !== "now" ? formatted + " ago" : formatted;
 }
 
+const formatRelativeMoment = (date: Date, now: Date): string => {
+  return moment(date).from(now);
+}
+
+const formatRelativeFast = (date: Date, now: Date): string => {
+  const msApart = Math.abs(now.getTime() - date.getTime());
+  const secondsApart = msApart / 1000;
+  if (secondsApart < 44) {
+    return "now";
+  } else if (secondsApart < 45*60) {
+    const minutes = Math.round(secondsApart/60.0);
+    return `${minutes}m`;
+  } else if (secondsApart < 22*60*60) {
+    const hours = Math.round(secondsApart/(60.0*60.0));
+    return `${hours}h`;
+  } else if (secondsApart < 26*24*60*60) {
+    const days = Math.round(secondsApart/(24*60*60.0));
+    return `${days}d`;
+  } else if (secondsApart < 335) {
```

Comment:
```suggestion
  } else if (secondsApart < 335*24*60*60) {
```

Reply:
Welp, this line definitely shows the need for a unit test.